quipucords/scanner/tasks.py

```diff
@@ -69,16 +70,94 @@ def get_task_runner_class(source_type, scan_type) -> type[ScanTaskRunner]:
     return getattr(get_scanner(source_type), runner_class_name)
 
 
+def set_scan_task_failure_on_exception(func):
+    """
+    Set the ScanTask's FAILURE status when handling unexpected exceptions.
+
+    This decorator assumes `scan_task_id` will be a keyword argument of the
+    function being wrapped.
+
+    Todo: Maybe merge/consolidate this with set_scan_job_failure_on_exception?
+    If we merge these two decorators, we probably need a clever way to determine
+    whether the relevant object to "fail" is a ScanJob or a ScanTask.
+    """
+
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs):
+        if not (scan_task_id := kwargs.get("scan_task_id", None)):
+            logger.warning(
+                f"Missing scan_task_id kwarg for decorated function {func.__name__}"
+            )
```

Comment:
same observation I made for scan_job decorator: https://github.com/quipucords/quipucords/pull/2768/commits/9907a930aa4414e9e5cb5547b6ec89037ca489e6#r1904196237